pkg/offsetguess/offsetguess.go

```diff
@@ -0,0 +1,338 @@
+// +build linux
+
+package offsetguess
+
+import (
+	"encoding/binary"
+	"fmt"
+	"math/rand"
+	"net"
+	"os"
+	"strconv"
+	"strings"
+	"syscall"
+	"unsafe"
+
+	"github.com/iovisor/gobpf/elf"
+
+	"github.com/weaveworks/tcptracer-bpf/pkg/byteorder"
+)
+
+type tcpTracerState uint64
+
+const (
+	threshold         = 200
+	thresholdInetSock = 2000
+)
+
+const (
+	uninitialized tcpTracerState = iota
+	checking
+	checked
+	ready
+)
+
+type guessWhat uint64
+
+const (
+	guessSaddr guessWhat = iota
+	guessDaddr
+	guessFamily
+	guessSport
+	guessDport
+	guessNetns
+	guessDaddrIPv6
+)
+
+type tcpTracerStatus struct {
+	status          tcpTracerState
+	pidTgid         uint64
+	what            guessWhat
+	offsetSaddr     uint64
+	offsetDaddr     uint64
+	offsetSport     uint64
+	offsetDport     uint64
+	offsetNetns     uint64
+	offsetIno       uint64
+	offsetFamily    uint64
+	offsetDaddrIPv6 uint64
+	err             byte
+	saddr           uint32
+	daddr           uint32
+	sport           uint16
+	dport           uint16
+	netns           uint32
+	family          uint16
+	daddrIPv6       [4]uint32
+}
+
+type fieldValues struct {
+	saddr     uint32
+	daddr     uint32
+	sport     uint16
+	dport     uint16
+	netns     uint32
+	family    uint16
+	daddrIPv6 [4]uint32
+}
+
+func listen(url, netType string, listenCompleted, closeListen chan struct{}) {
+	l, err := net.Listen(netType, url)
+	if err != nil {
+		panic(err)
```

Comment:
The previous comment about this panic was hidden for some reason https://github.com/weaveworks/tcptracer-bpf/pull/1#discussion_r97792777

It still needs to be addressed:

```
$ busybox nc -l 127.0.0.2  -p 9091
```
```
$ sudo ./tracer ../ebpf/tcptracer-ebpf.o 
panic: listen tcp4 127.0.0.2:9091: bind: address already in use
```